src/gen_caml.ml

```diff
@@ -268,10 +282,38 @@ let make_sql l =
   Buffer.add_string b ")";
   Buffer.contents b
 
+let in_var_module _label typ = Sql.Type.to_string typ
+
+let gen_in_substitution var =
+  if Option.is_none var.id.label then failwith "empty label in IN param";
+  sprintf {code| "(" ^ String.concat ", " (List.map T.Types.%s.to_literal %s) ^ ")"|code}
+    (in_var_module (Option.get var.id.label) var.typ)
+    (Option.get var.id.label)
+
+let extract_in_subst vars =
+  let rec extract acc = function
+    | [] -> List.rev acc
+    | Single _ :: tl -> extract acc tl
+    | SingleText p :: tl -> extract (`InVar p :: acc) tl
```

Comment:
this is backwards now, SingleText makes an impression of generic text substitution but generates specifically only `IN`